### PR TITLE
added User and Role entities, repositories,sevices, DataInitializer a…

### PR DESCRIPTION
**Pull request by less08**
…nd travis checkstyle plugin

### CODE REVIEW

File: src/main/java/com/example/bankservice/service/RoleServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    private final RoleRepository roleRepository;
+
+    @Override
+    public void add(Role role) {
+        roleRepository.save(role);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        return roleRepository.getRoleByRoleName((Role.RoleName.valueOf(roleName))).get();
```
Comment (vicire) [inline src/main/java/com/example/bankservice/service/RoleServiceImpl.java]: What happened if a role doesn`t exist in DB?

File: src/main/java/com/example/bankservice/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+public interface UserService {
+    void add(User user);
+
+    User getById(Long id);
+
+    User getByPhoneNumber(String phoneNumber);
+
+    void remove(Long id);
+}
```
Comment (vicire) [inline src/main/java/com/example/bankservice/service/UserService.java:13]: You also need to declare method `update`

File: src/main/java/com/example/bankservice/service/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private final UserRepository userRepository;
+
+    @Override
+    public void add(User user) {
+        userRepository.save(user);
+    }
+
+    @Override
+    public User getById(Long id) {
+        return userRepository.getUserById(id).get();
```
Comment (vicire) [inline src/main/java/com/example/bankservice/service/UserServiceImpl.java]: Same as in RoleService

**Review by vicire — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bankservice/service/UserService.java (hunk 2 above)
Comment (less08) [reply src/main/java/com/example/bankservice/service/UserService.java:13]: Not really, it's from the checklist "You can have a single save() method on the service layer for both ‘create’ and ‘update’ operations."
So maybe i'll just rename add() to save()

Comment (vicire) [reply src/main/java/com/example/bankservice/service/UserService.java:13]: Ok, it's optional but in my opinion, you can declare `add` and `update`. And they both will use `save` from repository

**Review by vicire — APPROVED** [review]

File: src/main/java/com/example/bankservice/model/User.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+@Data
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+    @Column(name = "date_of_birth")
+    private LocalDate dateOfBirth;
+    @Column(name = "phone_number")
+    private String phoneNumber;
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankservice/model/User.java:25]: please see checklist - this field should be unique

File: src/main/java/com/example/bankservice/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("select u from User u left join fetch u.roles where u.id =?1")
+    Optional<User> getUserById(Long id);
+
+    @Query("select u from User u left join fetch u.roles where u.phoneNumber =?1")
+    Optional<User> findByPhoneNumber(String phoneNumber);
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/example/bankservice/repository/UserRepository.java]: redundant empty line

File: src/main/java/com/example/bankservice/service/UserService.java (hunk 2 above)
Comment (KseniiaMakarova) [reply src/main/java/com/example/bankservice/service/UserService.java:13]: @vicire I agree with Lena on this point - there is no particular reason to create 2 method with the same realisation

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1 +1,4 @@
-
+spring.jpa.show-sql=true
+spring.jpa.open-in-view=false
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.h2.console.path=/h2-console
```
Comment (KseniiaMakarova) [inline src/main/resources/application.properties:4]: is console enabled by default?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**